test/write_test.py

```diff
@@ -70,3 +68,14 @@ async def test_a_large_event(event_loop):
         read_result = await c.get(stream_name, 0)
         print(read_result)
         assert read_result[0].event.type == "big_json"
+
+
+@pytest.mark.asyncio
+async def test_publish_raises_exception_if_not_authenticated(event_loop):
```

Comment:
before it was returning a message which, if you inspected it, said access denied.  but that felt confusing.  and inconsistent with the read APIs which do explicitly raise an exception when they see `AccessDenied`